corehq/apps/users/bulk_download.py

```diff
@@ -230,27 +228,29 @@ def parse_mobile_users(domain, user_filters, task=None, total_count=None):
         user_headers.extend(json_to_headers(
             {'location_code': list(range(1, max_location_length + 1))}
         ))
-    if is_multi_domain_download:
+    if is_cross_domain:
         user_headers += ['domain']
     return user_headers, get_user_rows(user_dicts, user_headers)
 
 
-def parse_web_users(domain, task=None, total_count=None):
+def parse_web_users(domain, user_filters, task=None, total_count=None):
     user_dicts = []
     max_location_length = 0
-    location_cache = LocationIdToSiteCodeCache(domain)
-    for n, user in enumerate(get_all_user_rows(domain, include_web_users=True, include_mobile_users=False,
-                                               include_inactive=False, include_docs=True)):
-        user_dict = make_web_user_dict(user, location_cache, domain)
-        user_dicts.append(user_dict)
-        max_location_length = max(max_location_length, len(user_dict["location_code"]))
-        if task:
-            DownloadBase.set_progress(task, n, total_count)
-    for m, invite in enumerate(Invitation.by_domain(domain)):
-        user_dict = make_invited_web_user_dict(invite, location_cache)
-        user_dicts.append(user_dict)
-        if task:
-            DownloadBase.set_progress(task, n + m, total_count)
+    (is_cross_domain, domains_list) = get_domains_from_user_filters(domain, user_filters)
+    progress = 0
+    for current_domain in domains_list:
+        location_cache = LocationIdToSiteCodeCache(current_domain)
+        for user in get_web_users_by_filters(current_domain, user_filters):
+            user_dict = make_web_user_dict(user, location_cache, current_domain)
+            user_dicts.append(user_dict)
+            max_location_length = max(max_location_length, len(user_dict["location_code"]))
```

Comment:
just confirming if this would happen,
```
>>> len(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
>>> len('')
0
```

Reply:
It shouldn't, `location_code` always gets populated: https://github.com/dimagi/commcare-hq/blob/123eedf5523878a71b0f8bebe7fe0cec2e2f3ef6/corehq/apps/users/bulk_download.py#L145

Reply:
not saying you added this but, location_code has location_codes, that is great :)